assets/src/design-system/components/contextMenu/index.js

```diff
@@ -0,0 +1,228 @@
+
+    a {
+      background-color: transparent;
+      text-decoration: none;
+    }
+
+    li {
+      a,
+      button,
+      div {
+        display: flex;
+        justify-content: space-between;
+        align-items: center;
+        padding: 8px 16px;
+        border: 0;
+        transition: background-color ${BUTTON_TRANSITION_TIMING};
+      }
+
+      span {
+        transition: color ${BUTTON_TRANSITION_TIMING};
+      }
+
+      a {
+        :active span,
+        :hover span {
+          color: ${theme.colors.fg.primary};
+        }
+      }
+
+      button {
+        width: 100%;
+        border-radius: 0;
+        background-color: transparent;
+
+        :disabled {
+          background-color: transparent;
+
+          span {
+            color: ${theme.colors.bg.tertiary};
+          }
+        }
+
+        :active span,
+        :hover span {
+          color: ${theme.colors.fg.primary};
+        }
+      }
+
+      &.separatorTop {
+        border-top: 1px solid ${theme.colors.bg.tertiary};
+      }
+
+      &.separatorBottom {
+        border-bottom: 1px solid ${theme.colors.bg.tertiary};
+      }
+
+      :hover a,
+      button:hover:not(:disabled) {
+        background-color: ${theme.colors.interactiveBg.secondaryHover};
+      }
+
+      :active a,
+      button:active:not(:disabled) {
+        background-color: ${theme.colors.interactiveBg.secondaryPress};
+      }
+    }
+  `
+);
+
+const ContextMenu = ({ isOpen, items, ...props }) => {
+  const [focusedIndex, setFocusedIndex] = useState(-1);
+  const listRef = useRef(null);
+  const ids = useMemo(() => items.map(() => uuidv4()), [items]);
+
+  /**
+   * Allow navigation of the list using the UP and DOWN arrow keys.
+   *
+   * @param {Event} ev The synthetic event
+   * @return {void} void
+   */
+  const handleKeyDown = useCallback(
+    (ev) => {
+      switch (ev.key) {
+        case DOWN:
+          ev.preventDefault();
+          if (listRef.current && focusedIndex < items.length - 1) {
+            let index = focusedIndex + 1;
+
+            while (index < items.length) {
+              const element = listRef.current?.children?.[index]?.children?.[0];
+              if (
+                FOCUSABLE_ELEMENTS.includes(element?.tagName) &&
+                !element?.disabled
+              ) {
+                setFocusedIndex(index);
+                return;
+              }
+              index++;
+            }
+          }
+          break;
+        case UP:
+          ev.preventDefault();
+          if (listRef.current && focusedIndex > -1) {
+            let index = focusedIndex - 1;
+
+            while (index > -1) {
+              const element = listRef.current?.children?.[index]?.children?.[0];
+              if (
+                FOCUSABLE_ELEMENTS.includes(element?.tagName) &&
+                !element?.disabled
+              ) {
+                setFocusedIndex(index);
+                return;
+              }
+              index--;
+            }
```

Comment:
a little bit of repeated code here but the direction is different. I could abstract this into a function and to pick a direction but I think it would make it less readable.

Thoughts?

Reply:
To be fair: it's already kind of hard to read.

This looks for the next focusable element ('a' or 'button'). If none are found, then it keeps the focus on the current element.

Reply:
We have a util actually to help with keyboard events. I know the current implementation has a switch like this, but the editor has a lot of focus traps and it'd probably be good to standardize how keyboard handlers are done? 

Something like this would clean it up - this works as is for nav, just not disabled items.
```
  const totalIndex = useMemo(() => items.length - 1, [items]);

  const handleKeyboardNav = useCallback(
    ({ key }) => {
      if ([isRTL ? 'ArrowDown' : 'ArrowUp'].includes(key)) {
        setFocusedIndex((currentIndex) =>
          currentIndex - 1 > -1 ? currentIndex - 1 : -1
        );
      }

      if ([isRTL ? 'ArrowUp' : 'ArrowDown'].includes(key)) {
        setFocusedIndex((currentIndex) =>
          currentIndex + 1 <= totalIndex ? currentIndex + 1 : currentIndex
        );
      }
    },
    [isRTL, totalIndex]
  );
  useKeyDownEffect(listRef, { key: ['down', 'up'] }, handleKeyboardNav, [
    handleKeyboardNav,
  ]);
```

Reply:
Chatted with brittany about this - we don't need to worry about RTL/LTR since it's not flipping vertically